src/Kestrel.Core/Internal/Http2/HPack/HPackDecoder.cs

```diff
@@ -110,9 +110,20 @@ internal HPackDecoder(int maxDynamicTableSize, DynamicTable dynamicTable)
 
         public void Decode(Span<byte> data, bool endHeaders, IHttpHeadersHandler handler)
         {
+            var headersObserved = false;
```

Comment:
This isn't the right place, the state needs to be preserved across calls to Decode so it can handle Continuation frames.